### PR TITLE
Appium v2 Support for wdio

### PR DESCRIPTION
**Pull request by pankaj443**
JIRA: https://browserstack.atlassian.net/browse/PAPP-457
Update WDIO flow to use execute viewportRect for getting deviceSize and pixel ratio.

### CODE REVIEW

File: percy/metadata/metadata.js [hunk 1]
```diff
@@ -16,12 +16,20 @@ class Metadata {
...
     this._statusBarHeight = statusBarHeight;
     this._navigationBarHeight = navigationBarHeight;
+    this._viewportRect = null;
   }
 
   // items that need caps are moved to getters as caps are not stored on wd driver object
   // So we need to make a lazy call to avoid making session get call in app automate context
   async caps() {
-    return await this.driver.getCapabilities();
+    /* istanbul ignore next */
```
Comment (ninadbstack) [inline percy/metadata/metadata.js:25]: why are we adding this on all if conditions ?

File: percy/metadata/metadata.js [hunk 2]
```diff
@@ -16,12 +16,20 @@ class Metadata {
...
+    if (typeof browser !== 'undefined') {
```
Comment (ninadbstack) [inline percy/metadata/metadata.js:26]: why are we accessing global browser here ? it shouldnt be directly accessed, we already have reference in this.driver

File: percy/metadata/metadata.js [hunk 3]
```diff
@@ -16,12 +16,20 @@ class Metadata {
...
+      /* eslint-disable */
```
Comment (ninadbstack) [inline percy/metadata/metadata.js:27]: whenever you disable linting rules please add a comment on why we need to do that

File: percy/metadata/iosMetadata.js [hunk 4]
```diff
@@ -6,14 +6,17 @@ class IosMetadata extends Metadata {
...
 
-    const caps = await this.caps();
     if (await this.staticData()) {
       const data = await this.staticData();
       return data.statusBarHeight * data.pixelRatio;
     }
 
-    // In Ios the height of statusBarHeight in caps needs to be multiplied by pixel ratio
-    return (caps.statBarHeight || 1) * (caps.pixelRatio || 1);
+    if (this.driver.wdio) {
```
Comment (pankaj443) [inline percy/metadata/iosMetadata.js:14]: kept it here since its ios specific

File: percy/metadata/iosMetadata.js [hunk 5]
```diff
@@ -26,16 +29,31 @@ class IosMetadata extends Metadata {
...
-    const caps = await this.caps();
     if (await this.staticData()) {
       const data = await this.staticData();
       return { width: data.screenWidth, height: data.screenHeight };
     }
-    const height = caps.statBarHeight * caps.pixelRatio + caps.viewportRect?.height;
-    const width = caps.viewportRect?.width;
+
+    var height, width;
+    if (this.driver.wdio) {
```
Comment (pankaj443) [inline percy/metadata/iosMetadata.js:38]: kept it here since its ios specific

File: percy/metadata/iosMetadata.js [hunk 6]
```diff
@@ -51,7 +69,15 @@ class IosMetadata extends Metadata {
   }
 
   async scaleFactor() {
-    return (await this.caps()).pixelRatio;
+    if (this.driver.wdio) {
```
Comment (pankaj443) [inline percy/metadata/iosMetadata.js:72]: kept it here since its ios specific

File: percy/metadata/metadata.js (hunk 1 above)
Comment (pankaj443) [reply percy/metadata/metadata.js]: removed, have only kept which are specific to ios

File: percy/metadata/metadata.js (hunk 2 above)
Comment (pankaj443) [reply percy/metadata/metadata.js]: updated to use driver.wdio

File: percy/metadata/metadata.js (hunk 3 above)
Comment (pankaj443) [reply percy/metadata/metadata.js]: not required anymore

File: percy/metadata/metadata.js (hunk 2 above)
Comment (ninadbstack) [reply percy/metadata/metadata.js]: No i mean we should do any driver specific things in the adaptor class [ the driverWrapper class ] and not in any other part of codebase.  you can update methods there

Comment (ninadbstack) [reply percy/metadata/metadata.js]: Also while this seems to rely on global browser object to be available, which is usually true in wdio, but its not necessary. Users can also write a script that does not use global browser [ and so adaptor checks for type of passed driver instead ]

Comment (pankaj443) [reply percy/metadata/metadata.js]: @ninadbstack yes i have moved caps in driverWrapper only, others which were specific to ios only I have kept in iosMetadata.

**Review by ninadbstack — APPROVED** [review]

**Review by ninadbstack — APPROVED** [review]

**Pull request merged**